### PR TITLE
Upgrade Webpacker 6 and fine-tune configuration

### PR DESCRIPTION
**Pull request by rserur**
# Fullstack PR Summary

[Asana Ticket](https://app.asana.com/0/1200734477019044/1201089129505921/f)

Also fixes [deployment issue ticket](https://app.asana.com/0/1200734477019044/1200810932677150).

## Included Changes:

- Upgraded webpacker to v6.0.0.rc.5
- Webpacker configuration and frontend application setup slightly refactored
- CSS & Sass moved into frontend
- CSS loader & Sass loader installed
- Changed to use ideas42 logo
- Details on Rails + React setup added to readme

## Notes to Reviewer
Looks like our [feature branch deployments](https://feature-build-pr-18.herokuapp.com/) have 500 errors to `api/v1/cats/index`-- looking at the application logs on Heroku, looks like an unrelated database issue that just means we need some additional Heroku configuration to properly set up the database? I'll create a separate ticket for it.

## Checklist
- [x] Have you updated documentation (if needed)?
- [x] Have you attached screenshots (if needed)?
- [x] Have you moved the Asana ticket to "In Review"?

### Browser Checking (latest versions)
- [x] Firefox
- [x] Chrome
- [x] Safari

## Before
image

## After 
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,6 +29,16 @@ Set environment variables for specific environments with `.env.development`, `.e
...
+### Rails and React
+
+We are using Rail's [Webpacker](https://github.com/rails/webpacker) to pre-process and bundle our React application. Webpacker coexists with the Rails asset pipeline, which is best used for server-side rendered pages. 
+
+Assets (images, fonts, and CSS) for our React application should be housed in our React scaffolding (`app/javascript`) unless a specific edge case requires otherwise. 
+
+Assets for server-side rendered pages (in context of the ideas42 Venture Studio, largely administrative pages) can be housed in the Rails asset pipeline (`app/assets`) in a conventional manner for Rails views (`app/views`).
+
+The React application is initially pulled in via a server-side render on `app/views/layouts/application.html.erb`. Webpacker pre-processes and bundles (see `config/webpacker.yml`) in concert with [Webpack.js](https://webpack.js.org/) itself (see `config/webpack/*` files) so that the application entrypoint code (`app/javascript/entrypoints/application.js`) can be served.
+
```
Comment (amandamunoz) [inline README.md:41]: 🔥

File: app/javascript/entrypoints/application.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// Rails-related libraries
```
Comment (amandamunoz) [inline app/javascript/entrypoints/application.js:1]: love the comments here to help delineate

File: config/webpacker.yml [hunk 3]
```diff
@@ -57,21 +24,26 @@ development:
...
     host: localhost
     port: 3035
-    public: localhost:3035
+    # Hot Module Replacement updates modules while the application is running without a full reload
     hmr: false
-    # Inline should be set to true if using HMR
-    inline: true
-    overlay: true
+    client:
+      # Should we show a full-screen overlay in the browser when there are compiler errors or warnings?
```
Comment (amandamunoz) [inline config/webpacker.yml:30]: do we want to create follow-up tickets to address the questions here?

**Review by amandamunoz — APPROVED** [review]
looks great, thanks for the clear comments 💯 thanks for creating a followup ticket for the db error

File: config/webpacker.yml (hunk 3 above)
Comment (rserur) [reply config/webpacker.yml:30]: Oh-- these were autogenerated comments by Webpacker-- I think they're just meant to be descriptive and the defaults here are good/useful for development purposes. I'll reword them so we know these aren't actually questions to be asked.

**Pull request merged**